### PR TITLE
Text Form Bug Regressions

### PR DESCRIPTION
**Pull request by ctsims**
Regression tests for bugs caught during 2.22 QA

### CODE REVIEW

File: unit-tests/src/org/commcare/android/tests/processing/KeyRecordTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/**
+ * 
+ */
```
Comment (phillipm) [inline unit-tests/src/org/commcare/android/tests/processing/KeyRecordTest.java]: Please turn off the setting that adds these comments to files

File: unit-tests/src/org/commcare/android/tests/processing/KeyRecordTest.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    //TODO: Move this to the application or somewhere better static
+    static LivePrototypeFactory factory = new LivePrototypeFactory();
+
+    @Before
+    public void setupTests() {
+        TestUtils.initializeStaticTestStorage();
+    }
+    
+    @Test
+    public void testKeyRecordParse() {
```
Comment (phillipm) [inline unit-tests/src/org/commcare/android/tests/processing/KeyRecordTest.java]: A comment detailing what functionality this is testing would be great. What are keys? When does this happen in the app's logic? Do we expect to be in a particular state before or after a key parse? I know I could go read the code and figure all this stuff out, but a few lines of text here would be a much easier way of giving me that context.

File: unit-tests/src/org/commcare/android/tests/processing/KeyRecordTest.java (hunk 1 above)
Comment (ctsims) [inline unit-tests/src/org/commcare/android/tests/processing/KeyRecordTest.java]: Done. Fixing this one in new commit

File: unit-tests/src/org/commcare/android/tests/processing/FormStorageTest.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        FormDef def = XFormUtils.getFormFromResource("/resources/forms/placeholder.xml");
+        try {
+            ExtUtil.deserialize(ExtUtil.serialize(def), FormDef.class, TestUtils.factory);
+        } catch (IOException e) {
+            e.printStackTrace();
+            throw new RuntimeException(e.getMessage());
+        } catch (DeserializationException e) {
+            e.printStackTrace();
+            throw new RuntimeException(e.getMessage());
+        }
```
Comment (phillipm) [inline unit-tests/src/org/commcare/android/tests/processing/FormStorageTest.java]: These 2 catch branches are identical, please merge them:
} catch (IOException | DeserializationException e) {

File: unit-tests/src/org/commcare/android/tests/processing/FormStorageTest.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
+/**
+ * 
+ */
+package org.commcare.android.tests.processing;
+
+import java.io.IOException;
+
+import org.commcare.android.junit.CommCareTestRunner;
+import org.commcare.android.shadows.SQLiteDatabaseNative;
+import org.commcare.android.util.LivePrototypeFactory;
```
Comment (phillipm) [inline unit-tests/src/org/commcare/android/tests/processing/FormStorageTest.java]: Unused import

File: unit-tests/src/org/commcare/android/tests/processing/FormStorageTest.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
+/**
+ * 
+ */
```
Comment (phillipm) [inline unit-tests/src/org/commcare/android/tests/processing/FormStorageTest.java]: Useless comment block here too.

File: unit-tests/src/org/commcare/android/tests/processing/KeyRecordTest.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
+package org.commcare.android.tests.processing;
+
+import static junit.framework.Assert.assertEquals;
```
Comment (phillipm) [inline unit-tests/src/org/commcare/android/tests/processing/KeyRecordTest.java]: Unused import...

File: unit-tests/src/org/commcare/android/tests/processing/KeyRecordTest.java (hunk 2 above)
Comment (ctsims) [inline unit-tests/src/org/commcare/android/tests/processing/KeyRecordTest.java]: Added some comments. This test setup isn't very thorough, just wanted to make sure we had basic regressions for the XML processing setup.

Added a TODO as well for what tests should be in the next pass

File: unit-tests/src/org/commcare/android/tests/processing/KeyRecordTest.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+import org.commcare.android.database.user.models.ACase;
+import org.commcare.android.junit.CommCareTestRunner;
+import org.commcare.android.shadows.SQLiteDatabaseNative;
+import org.commcare.android.util.LivePrototypeFactory;
+import org.commcare.android.util.TestUtils;
+import org.commcare.cases.model.Case;
```
Comment (phillipm) [inline unit-tests/src/org/commcare/android/tests/processing/KeyRecordTest.java]: Unused

File: unit-tests/src/org/commcare/android/tests/processing/KeyRecordTest.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
+package org.commcare.android.tests.processing;
+
+import static junit.framework.Assert.assertEquals;
+
+import org.commcare.android.database.user.models.ACase;
```
Comment (phillipm) [inline unit-tests/src/org/commcare/android/tests/processing/KeyRecordTest.java]: Unused

**Comment by phillipm** [comment]
Looks good, though please run linter before submitting PRs

**Pull request merged**